src/main/java/org/helpberkeley/memberdata/WorkflowExporter.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) 2024 helpberkeley.org
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+package org.helpberkeley.memberdata;
+
+import java.util.*;
+
+public class WorkflowExporter extends Exporter {
+
+    WorkflowBean bean;
+    private StringBuilder updateWarnings;
+    private WorkflowParser parser;
+    private List<WorkflowBean> updatedBeans;
+    private int numMembers;
+
+    public WorkflowExporter(WorkflowParser parser) {
+        this.parser = parser;
+        updatedBeans = new ArrayList<WorkflowBean>();
+        updateWarnings = new StringBuilder("| UserName | Name | Phone | Phone 2 | Neighborhood | City | Address | Condo | Details |\n");
+        updateWarnings.append("|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|:---:|\n");
+        numMembers = 0;
+    }
+
+    public String getWarnings() { return updateWarnings.toString(); }
+
+    public String updateMemberData(Map<String, User> users, Map<String, DetailsPost> deliveryDetails) {
+        String errors = "";
+
+        while ((bean = parser.nextRow()) != null) {
+            if (parser.isMemberRow(bean)) {
+                numMembers++;
+                if (numMembers > (Constants.AVG_RUN_SIZE*10)) {
+                    throw new MemberDataException("This is many more members [" + numMembers +
+                            "] than the typical run, did you accidentally upload a sheet with all current members?");
```

Comment:
this is not tested yet, the only way I can think of to test it is to create 300 fake users - is that what I should do?

Reply:
One technique used to solve this problem is to allow the test to change the limit.  See examples of this with the usage of ApiClient.RETRY_NAP_MILLISECONDS in ApiClientTest.java, where I didn't want the negative tests to take the default many seconds of timeout time.  Note, that I was a little lazy here and didn't fully encapsulate this.  To do this cleaner the member that needs to be modified should be private, ideally not static, and accessed through getters/setters for setting and restoring.  But often one needs to make it static because the test code doesn't have access to the instance of the class at test runtime.